kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java

```diff
@@ -7015,11 +7015,42 @@ protected void handleAccessResource(HttpServletRequest req, HttpServletResponse
 					return; 
 				}
 
+				// If there is a direct link to the asset, no sense streaming it.
+				// Send the asset directly to the load-balancer or to the client
+				URI directLinkUri = m_storage.getDirectLink(resource);
```

Comment:
We just fixed this up. There was a cast to BaseResourceEdit failing and getting translates into a silent 404. I think Sam will update the pull request today.

> On Feb 18, 2016, at 4:58 AM, Matthew Buckett <notifications@github.com> wrote:
> 
> In kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java:
> 
> > @@ -7015,11 +7015,42 @@ protected void handleAccessResource(HttpServletRequest req, HttpServletResponse
> >  					return; 
> >  				}
> >  
> > +				// If there is a direct link to the asset, no sense streaming it.
> > +				// Send the asset directly to the load-balancer or to the client
> > +				URI directLinkUri = m_storage.getDirectLink(resource);
> @ottenhoff What do you mean by break?
> 
> The Content-Length header will be wrong if you're not reading from the InputStream and aren't redirecting them directly to the content. As with a WrappedContentResource extends the content by adding on a HTML header/footer which means the content length needs increasing. I think a browser will hang around waiting for the extra data which is never going to come.
> 
> —
> Reply to this email directly or view it on GitHub.
>